pytorch_lightning/core/lightning.py

```diff
@@ -1692,4 +1693,20 @@ def hparams(self) -> Union[AttributeDict, str]:
 
     @hparams.setter
     def hparams(self, hp: Union[dict, Namespace, Any]):
-        self.save_hyperparameters(hp, frame=inspect.currentframe().f_back.f_back)
+        hparams_assignment_name = self.__get_hparams_assignment_variable()
+        self._hparams_name = hparams_assignment_name
+        self._set_hparams(hp)
+
+    def __get_hparams_assignment_variable(self):
+        """
+        looks at the code of the class to figure out what the user named self.hparams
+        this only happens when the user explicitly sets self.hparams
+        """
+        class_code = inspect.getsource(self.__class__)
+        lines = class_code.split('\n')
+        for line in lines:
+            line = re.sub(r"\s+", "", line, flags=re.UNICODE)
+            if 'self.hparams=' in line:
```

Comment:
this is a naive hack assuming user uses standard naming, it will fail on 
```
class AAA(...):
    def __init__(myself, param1, ...):
         self.hparams = param1
```

Reply:
nope... 'hparams' is a fallback... try it!

Reply:
the point is with rename `self` to `me` or what ever

Reply:
also what about spacing
```
self.hparams                <50x spaces> = param1
```
or rename the vasiable
```
class AAA():
  def __init__(me, param1):
    copy_param = copy.deepcopy(param1)
    me.hparams = copy_param
```